Orange/widgets/model/owrules.py

```diff
@@ -323,14 +323,18 @@ def update_model(self):
         learner is used within this widget (for example, it must not be
         accessed from test&score widget).
         """
+        self.clear_messages()
         if self.check_data():
-            with self.progressBar() as progress:
-                self.learner.set_progress_advance_callback(progress.advance)
-                self.model = self.learner(self.data)
-                self.learner.clear_progress_advance_callback()
-            self.model.name = self.learner_name
-            self.model.instances = self.data
-            self.valid_data = True
+            if self.data.is_sparse():
+                self.Error.sparse_not_supported()
```

Comment:
The idea was to have Learner (subtypes) define `supports_sparse` such as they have (and possibly use) [`supports_multiclass` and `supports_weights`](https://github.com/kernc/orange3/blob/2b1398fb5b1b2e15bb83ef7eefa9212f8f336b1e/Orange/base.py#L78-L79). The learners that don't support sparse data then define `supports_sparse = False` (or vice versa), and the handling of state where sparse data is erroneously passed to a unsupporting learner widget is handled in a single place in `OWBaseLearner`.